jenkins/PerfCI_Chaos/03_PerfCI_Choas_VMs_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,197 @@
+        GOOGLE_CREDENTIALS_FILE = 'credentials.json'
+        GOOGLE_TOKEN_FILE = 'token.json'
+        GOOGLE_DESTINATION_PATH = '/tmp'
+        GOOGLE_DRIVE_SHARED_DRIVE_ID = credentials('perfci_google_drive_shared_drive_id')
+        RUN_ARTIFACTS_URL = credentials('perfci_run_artifacts_url')
+        WINDOWS10_URL = credentials('perfci_windows10_url')
+        WINDOWS11_URL = credentials('perfci_windows11_url')
+        QUAY_BENCHMARK_RUNNER_REPOSITORY = credentials('perfci_quay_repository')
+        WINDOWS_SERVER_2019_URL = credentials('perfci_windows_server_2019_url')
+        WINDOWS_SERVER_2022_URL = credentials('perfci_windows_server_2022_url')
+        KUBEADMIN_PASSWORD = readFile('/home/jenkins/.kube/kubeadmin-password').trim()
+        KUBEADMIN_PASSWORD_PATH = '/home/jenkins/.kube/kubeadmin-password'
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        WORKSPACE='/home/jenkins'
+        CONTAINER_PRIVATE_KEY_PATH = '/root/.ssh/provision_private_key'
+        SAVE_ARTIFACTS_LOCAL = 'False'
+        ENABLE_PROMETHEUS_SNAPSHOT = 'True'
+        DELETE_ALL = 'False' // Not delete the running Windows11 VMs
+        RUN_STRATEGY = 'True' // runStrategy: Always for NHC/ SNR operators
+        RUN_TYPE = 'chaos_ci'
+        BOOTSTORM_SCALE = 80
+        THREADS_LIMIT= 20
+        TIMEOUT = 8000
+        WORKSPACE_PATH = "${env.WORKSPACE}/workspace/${env.JOB_NAME}/"
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+                        sh '''
+                            # Check if the image exists
+                            if [[ "$(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)" != "" ]]; then
+
+                                # Get containers using the image
+                                containers=$(sudo podman ps -a --filter ancestor=${QUAY_BENCHMARK_RUNNER_REPOSITORY} -q)
+
+                                if [[ "$containers" != "" ]]; then
+                                    echo "Stopping and removing containers using the image..."
+                                    sudo podman stop $containers
+                                    sudo podman rm -f $containers
+                                fi
+
+                                # Now remove the image
+                                sudo podman rmi -f $(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)
+                            fi
+                        '''
+                }
+            }
+        }
+
+        stage('⚙️ SET SSH key') {
+            steps {
+                script {
+                    sh 'mkdir -p $WORKSPACE/.ssh/'
+                    sh 'sudo chmod 700 $WORKSPACE/.ssh/'
```

Comment:
1. If you can create the directory, it's not necessary (or appropriate) to use sudo here.
2. Need to quote "$WORKSPACE"
3. Do this in one step (also avoids race condition attack):
```
sh 'umask 700 && mkdir -p "$WORKSPACE"/.ssh'

Reply:
Fixed